src/runtime/disco/builtin.h

```diff
@@ -52,6 +52,12 @@ NDArray DiscoEmptyNDArray(ShapeTuple shape, DataType dtype, Device device);
  * \return The outcome of allreduce
  */
 void AllReduce(NDArray send, ReduceKind reduce_kind, NDArray recv);
+/*!
+ * \brief Perform an allgather operation using the underlying communication library
+ * \param send The array send to perform allgather on
+ * \return The outcome of allgather
+ */
+void AllGather(NDArray send, Optional<NDArray> recv);
```

Comment:
The signature here doesn't match the signature of the implementation.  In the implementation, the second parameter is `NDArray recv` and not `Optional<NDArray> recv`.